### PR TITLE
Updates to palaeorotate()

### PR DESCRIPTION
**Pull request by LewisAJones**
This is a first draft of an update to the `palaeorotate()` function. The aim of this PR is to:

- Remove mantle reference frame/hybrid Global Plate Models as they are not recommended for reconstructing 'true' palaeocoordinates
- Make the TorsvikCocks2017 Global Plate Model available to users
- Enable users to make use of the new reconstruction grids which have a temporal resolution of 1 million years (equivalent to the GPlates API) and are fully peer-reviewed and documented

Alongside doing this, I've tried to improve the function code and the (horribly written) tests.

@willgearty and @Buffan3369 I think you're both best positioned to review this, but let me know if you don't have time and I can try to find another reviewer.

Note: the GPlates Web Service seems to currently be having some issues, so this PR will likely fail R CMD Checks on occasion. To try to prevent the API being overloaded with requests, I've added a rest between the GET requests (this seems to be helping).

### CODE REVIEW

File: R/palaeorotate.R [hunk 1]
```diff
@@ -1,41 +1,43 @@
 #' Palaeorotate fossil occurrences
 #'
-#' A function to generate palaeocoordinates for fossil occurrence data
+#' A function to estimate palaeocoordinates for fossil occurrence data
```
Comment (willgearty) [inline R/palaeorotate.R:3]: 👍

File: R/palaeorotate.R [hunk 2]
```diff
@@ -423,43 +362,34 @@ palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
...
+                  GOLONKA = query)
     # Define GPlates model names
-    query[["MULLER2022"]]$model <- "MULLER2022"
     query[["MERDITH2021"]]$model <- "MERDITH2021"
-    query[["MULLER2019"]]$model <- "MULLER2019"
-    query[["MULLER2016"]]$model <- "MULLER2016"
     query[["PALEOMAP"]]$model <- "PALEOMAP"
+    query[["TorsvikCocks2017"]]$model <- "TorsvikCocks2017"
+    # anchor plate id 1 required for palaeomagnetic reference frame
+    query$TorsvikCocks2017$anchor_plate_id <- 1
```
Comment (willgearty) [inline R/palaeorotate.R:381]: Not required for MATTHEWS2016_pmag_ref?

**Review by willgearty — APPROVED** [review]
Just some very minor comments/questions, looks great to me otherwise!

File: R/palaeorotate.R (hunk 2 above)
Comment (LewisAJones) [reply R/palaeorotate.R:381]: Oddly, MATTHEWS2016_pmag_ref and MATTHEWS2016_mantle_ref are implemented as two separate endpoints, and the anchor_plate_id does not need defining based on the [GWS docs](https://gwsdoc.gplates.org/models#matthews2016_pmag_ref).

File: R/palaeorotate.R [hunk 3]
```diff
@@ -211,60 +173,53 @@ palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
...
-  # Add warnings for use of mantle reference frame models
-  if (any(model %in% c("MULLER2022", "MULLER2019",
-                   "MATTHEWS2016_mantle_ref"))) {
-    warning(paste0("Selected model(s) use a mantle reference frame and are ",
-    "not recommended for reconstructing palaeocoordinates. See details."))
+  # Add stop for removed models
+  if (any(model %in% c("MULLER2022", "MULLER2019", "MULLER2016",
+                       "MATTHEWS2016_mantle_ref", "SETON2012"))) {
+    stop(paste0("Selected model(s) have recently been removed as they are not",
+    " in a palaeomagnetic reference frame. See details for available models."))
```
Comment (Buffan3369) [inline R/palaeorotate.R:217]: I would maybe suggest displaying the problematic models. One possibility:
```
deprecated_models <- c("MULLER2022", "MULLER2019", "MATTHEWS2016_mantle_ref", "SETON2012")
  if (any(model %in% deprecated_models)) {
paste0("Selected model(s) have recently been removed as they are not",
       " in a palaeomagnetic reference frame: ", 
       paste(deprecated_models[which(model %in% deprecated_models)], collapse = ", "),
       ". See details for available models.")
}
```

File: R/palaeorotate.R [hunk 4]
```diff
@@ -353,17 +291,10 @@ palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
...
-    # Get available rotation ages
-    rot_age <- colnames(ref_model)[4:ncol(ref_model)]
-    rot_age <- unique(as.numeric(sub(".*_", "", rot_age)))
-
     # Calculate rotation ages for data
-    occdf$rot_age <- rot_age[sapply(seq_len(nrow(occdf)), function(i) {
-        which.min(abs(occdf[i, age] - rot_age))
-      })]
+    occdf$rot_age <- round(occdf[, age], digits = 0)
+    # Set 0 to 1
```
Comment (Buffan3369) [inline R/palaeorotate.R]: doesn't work with zeros?

**Review by Buffan3369 — APPROVED** [review]
Nice job Lewis, just like Will, just a minor comment, everything works perfectly otherwise!

File: R/palaeorotate.R (hunk 3 above)
Comment (LewisAJones) [reply R/palaeorotate.R:217]: Yeah, that's fair enough! Thanks!

File: R/palaeorotate.R (hunk 4 above)
Comment (LewisAJones) [reply R/palaeorotate.R]: Sorry 😬 can you be a bit more specific? What doesn't work with zeros? The round function?

Comment (LewisAJones) [reply R/palaeorotate.R]: Ahh it looks like you might have been reviewing an earlier commit and not the most recent one?

Comment (willgearty) [reply R/palaeorotate.R]: Just checked, and this seems to work properly for zeros.

**Comment by LewisAJones** [comment]
Thanks for the review @willgearty and @Buffan3369!

File: R/palaeorotate.R (hunk 4 above)
Comment (Buffan3369) [reply R/palaeorotate.R]: Yes sorry Lewis, just checked and in worked properely ^^'.

**Pull request merged**